Packs/DeCYFIR/Integrations/decyfir/decyfir.py

```diff
@@ -0,0 +1,479 @@
+
+            # malicious-mobile-apps
+            if json_data := decyfir_alerts_incidents.get(VAR_MALICIOUS_MOBILE_APPS):
+                incidents_json_data = self.prepare_incidents_for_digital_risk(json_data, LABEL_DIGITAL_RISK_S_PE,
+                                                                              LABEL_MALICIOUS_MOBILE_APPS)
+                return_data = return_data + incidents_json_data
+
+            # confidential-files
+            if json_data := decyfir_alerts_incidents.get(VAR_CONFIDENTIAL_FILES):
+                incidents_json_data = self.prepare_incidents_for_digital_risk(json_data, LABEL_DIGITAL_RISK_S_PE,
+                                                                              LABEL_CONFIDENTIAL_FILES)
+                return_data = return_data + incidents_json_data
+
+            # dumps-pii-cii
+            if json_data := decyfir_alerts_incidents.get(VAR_DUMPS_PII_CII):
+                incidents_json_data = self.prepare_incidents_for_digital_risk(json_data, LABEL_DIGITAL_RISK_S_PE,
+                                                                              LABEL_DUMPS_PII_CII)
+                return_data = return_data + incidents_json_data
+
+            return return_data
+        except Exception as e:
+            raise DemistoException(str(e))
+
+
+# commands
+# This is the call made when pressing the integration Test button.
+def test_module(client, decyfir_api_key):
+    url = f"{PROD_API_PATH}" + f"/{VAR_ATTACK_SURFACE}?" + f"type={VAR_OPEN_PORTS}" \
+          + "&size=1" + "&key=" + f"{decyfir_api_key}"
+    response = client._http_request(
+        full_url=url,
+        method='GET',
+        resp_type='response')
+
+    if response.status_code == 200:
+        return 'ok'
+    elif response.status_code == 401 or response.status_code == 403:
+        return 'Not Authorized'
+    else:
+        return f"Error_code: {response.status_code}" + ", Please contact the DeCYFIR team to assist you further on this."
+
+
+def fetch_incidents(client, last_run, first_fetch, decyfir_api_key, incident_type, max_fetch):
+    try:
+        start_fetch = dateparser.parse(last_run.get("last_fetch")) if last_run else dateparser.parse(first_fetch)
+        start_fetch_timestamp_val: float = start_fetch.timestamp() if isinstance(start_fetch, datetime) else 0.0
+
+        start_fetch_timestamp: int = int(start_fetch_timestamp_val * 1000)
+
+        # To get the DeCYFIR data in JSON format
+        json_decyfir_data = client.get_decyfir_data(after_val=start_fetch_timestamp,
+                                                    decyfir_api_key=decyfir_api_key,
+                                                    incident_type=incident_type, max_fetch=max_fetch)
+
+        decyfir_incidents = client.convert_decyfir_data_to_incidents_format(json_decyfir_data)
+
+        # Assigning the current date time value to last_fetch for next run
+        last_fetch_time = datetime.now().strftime(DATE_FORMAT)
```

Comment:
It is better to take the last incident's timestamp and not the current time since If an incident was created but not yet fetched, or using the limit got only some of the incidents available, the rest will be lost (as they are older than right now).

Reply:
In this case we have different, will explain you in call.